File: js/src/2341-numberOfPairs.js
/**
 * @param {number[]} nums
 * @return {number[]}
 * 判断一个数组中有几个数对，有几个单独的数
 */
const numberOfPairs = function(nums) {
  const dict = {};
  let pair = 0;
  let remain = 0;
  nums.forEach((num) => {
    if (dict[num]) {
      dict[num] = null;
      pair++;
      remain--;
    } else {
      dict[num] = true;
      remain++;
    }
  });
  return [pair, remain];
};

export { numberOfPairs };

File: ruby/src/2341-number-of-pairs.rb
# @param {Integer[]} nums
# @return {Integer[]}
def number_of_pairs(nums)
  dict = {};
  pair = 0;
  remain = 0;
  nums.each do |i|
    if dict[i]
      dict[i] = nil;
      pair = pair + 1;
      remain = remain - 1;
    else
      dict[i] = true;
      remain = remain + 1;
    end
  end
  return [pair, remain];
end
